rocAL_pybind/rocal_pybind.cpp

```diff
@@ -155,6 +160,130 @@ std::unordered_map<int, std::string> rocalToPybindOutputDtype = {
     {5, "int32"},
 };
 
+#if ENABLE_DLPACK
+    //dlpack functions
+    static bool supported_dl_device_type(const DLDeviceType &devType) {
+        switch (devType) {
+            case kDLROCM:
+            case kDLROCMHost:
+            case kDLCPU:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    static RocalTensorOutputType get_data_type(const DLDataType &dtype) {
+        if (dtype.lanes != 1) {
+            throw std::runtime_error("Data type lanes != 1 is not supported.");
+        }
+
+        if (dtype.bits == 8) {
```

Comment:
I think it is better to use switch in the outer loop and if else in the inner loop